spec/lib/data_magic/config_spec.rb

```diff
@@ -31,23 +31,36 @@
       fake_s3 = class_spy("Fake Aws::S3::Client")
       fake_get_object_response = double(
         "S3 response",
-        body: StringIO.new({ 'index' => 'fake-index' }.to_yaml)
+        body: StringIO.new({ 'index' => 'fake-index' }.to_yaml),
+        isOK: true,
+        status: 200
       )
-      fake_list_objects_response = double(
-        "S3 response",
-        contents: [double("item", key: "data.yaml")]
-      )
-
       allow(fake_s3).to receive(:get_object)
         .with(bucket: 'mybucket', key: 'data.yaml')
         .and_return(fake_get_object_response)
-      allow(fake_s3).to receive(:list_objects)
-        .with(bucket: 'mybucket')
-        .and_return(fake_list_objects_response)
       config = DataMagic::Config.new(s3: fake_s3)
       expect(config.s3).to eq(fake_s3)
       expect(config.data["index"]).to eq("fake-index")
     end
+
+    it "raises error if s3 errors" do
```

Comment:
This is a great test. Can you do another one for getting 404s with ALLOW_MISSING_YML unset?